butterknife-compiler/src/test/java/butterknife/UnbinderTest.java

```diff
@@ -190,68 +215,326 @@
+                "    View view1;",
+                "    Unbinder(T target) {",
+                "      super(target);",
+                "    }",
+                "    @Override protected void unbind(T target) {",
+                "      super.unbind(target);",
+                "      view1.setOnClickListener(null);",
+                "      target.unbinder2 = null;",
+                "    }",
+                "  }",
+                "}"
+            ));
+
+    assertAbout(javaSource()).that(source)
+        .processedWith(new ButterKnifeProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expectedSource1, expectedSource2);
+  }
+
+  @Test public void childUsesOwnUnbinder() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.Test",
+        Joiner.on('\n')
+            .join(
+                "package test;",
+                "import android.support.v4.app.Fragment;",
+                "import butterknife.ButterKnife;",
+                "import butterknife.OnClick;",
+                "import butterknife.Unbinder;",
+                "public class Test extends Fragment {",
+                "  @Unbinder ButterKnife.Unbinder unbinder;",
+                "  @OnClick(1) void doStuff1() { }",
+                "}",
+                "class TestOne extends Test {",
+                "  @OnClick(1) void doStuff2() { }",
+                "}"
+            ));
+
+    JavaFileObject expectedSource1 = JavaFileObjects.forSourceString("test/Test$$ViewBinder",
+        Joiner.on('\n')
+            .join(
+                "package test;",
+                "import android.view.View;",
+                "import butterknife.internal.DebouncingOnClickListener;",
+                "import butterknife.internal.Finder;",
+                "import butterknife.internal.ViewBinder;",
+                "import java.lang.IllegalStateException;",
+                "import java.lang.Object;",
+                "import java.lang.Override;",
+                "import java.lang.SuppressWarnings;",
+                "public class Test$$ViewBinder<T extends Test> implements ViewBinder<T> {",
+                "  @Override public void bind(final Finder finder, final T target, Object source) {",
+                "    Unbinder unbinder = createUnbinder(target);",
+                "    View view;",
+                "    view = finder.findRequiredView(source, 1, \"method 'doStuff1'\");",
+                "    unbinder.view1 = view;",
+                "    view.setOnClickListener(new DebouncingOnClickListener() {",
+                "      @Override public void doClick(View p0) {",
+                "        target.doStuff1();",
+                "      }",
+                "    });",
+                "    target.unbinder = unbinder;",
+                "  }",
+                "  @SuppressWarnings(\"unchecked\")",
+                "  protected <U extends butterknife.ButterKnife.Unbinder<T>> U createUnbinder(T target) {",
```

Comment:
Tried to play with this again, and code wise it doesn't make any difference in relation to the casts, except that in your case every child class has to override a method with `<U extends Parent$$ViewBinder.Unbinder<T>>` (and that including second/third and so on generations) instead of `<U extends ButterKnife.Unbinder<T>>`. And still cast binder to `U` inside the methods. 
Also using the generated `Unbinder` will result in one additional layer of complexity, since, we will have to know the name of the initial parent, this will require an additional loop / or field in the `BindingClass`